core/range/reverse_each_spec.rb

```diff
@@ -0,0 +1,29 @@
+require_relative '../../spec_helper'
+
+ruby_version_is "3.3" do
+  describe "Range#reverse_each" do
+    it "works efficiently for very long Ranges of Integers" do
+      (1..2**100).reverse_each.take(3).size.should == 3
+    end
```

Comment:
Not sure if this test fails in case `#reverse_each` works inefficiently.

Reply:
This test is very much skirting the (MRI) implementation.

A Range is pretty much just a combination of a begin value, an end value, and an operation to get to the next value. In MRI 3.2 and below, there is no way to get to the previous value, so this code creates an array of `2**100` elements, then takes the last three elements of this array. So in MRI 3.2, this will result in either a `NoMemoryError` or an extremely long evaluation/timeout.
MRI 3.3 added a specialization for this method when using Integer values, this code now boils down to something like `[2**100, (2**100).pred, (2**100).pred.pred]` and finishes "instantly"
This specialization currently only exists for Integer values, that's why there I've added a spec for the a-z-Range as well, to ensure other types still work.